enterprise/cmd/precise-code-intel-worker/internal/worker/handler.go

```diff
@@ -270,6 +279,16 @@ func writeData(ctx context.Context, lsifStore LSIFStore, id int, groupedBundleDa
 	return nil
 }
 
+func isUniqueConstraintViolation(err error) bool {
+	for ex := err; ex != nil; ex = errors.Unwrap(ex) {
+		if pgErr, ok := ex.(*pgconn.PgError); ok {
+			return pgErr.Code == "23505"
+		}
+	}
+
+	return false
```

Comment:
```suggestion
	var pgErr *pgconn.PgError
	if errors.As(err, &pgErr) {
		return pgErr.Code == "23505"
	}
	
	return false
```

Think this should do the trick :slightly_smiling_face:

Reply:
You are correct.